torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -18,6 +17,46 @@ Tensor L1LossImpl::forward(const Tensor& input, const Tensor& target) {
 
 // ============================================================================
 
+KLDivLossImpl::KLDivLossImpl(const KLDivLossOptions& options_)
+    : options(options_) {}
+
+void KLDivLossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::KLDivLoss";
+}
+
+Tensor KLDivLossImpl::forward(const Tensor& input, const Tensor& target) {
+  return torch::kl_div(input, target, options.reduction());
+}
+
+// ============================================================================
+
+MSELossImpl::MSELossImpl(const MSELossOptions& options_) : options(options_) {}
+
+void MSELossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::MSELoss";
+}
+
+Tensor MSELossImpl::forward(const Tensor& input, const Tensor& target) {
+  return torch::mse_loss(input, target, options.reduction());
+}
+
+// ============================================================================
+
+BCELossImpl::BCELossImpl(const BCELossOptions& options_) : options(options_) {
+  register_parameter("weight", weight);
+}
+
+void BCELossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::BCELoss";
+}
+
+Tensor BCELossImpl::forward(const Tensor& input, const Tensor& target) {
+  return torch::binary_cross_entropy(
+      input, target, weight, options.reduction());
```

Comment:
It would be fantastic to add `F::binary_cross_entropy` in this PR, and here we can call `F:: binary_cross_entropy`, to match the Python version better. :)